Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -602,29 +653,41 @@ void CGContextAddLines(CGContextRef context, const CGPoint* points, unsigned cou
  @Notes
 */
 CGPathRef CGContextCopyPath(CGContextRef context) {
-    NOISY_RETURN_IF_NULL(context, StubReturn());
-    UNIMPLEMENTED();
-    return StubReturn();
+    NOISY_RETURN_IF_NULL(context, nullptr);
+
+    if (!context->HasPath()) {
+        return nullptr;
+    }
+
+    return CGPathCreateCopy(context->Path());
 }
 
 /**
  @Status Stub
  @Notes
 */
 CGPoint CGContextGetPathCurrentPoint(CGContextRef context) {
-    NOISY_RETURN_IF_NULL(context, StubReturn());
-    UNIMPLEMENTED();
-    return StubReturn();
+    NOISY_RETURN_IF_NULL(context, CGPointZero);
+
+    if (!context->HasPath()) {
+        return CGPointZero;
+    }
+
+    return CGPathGetCurrentPoint(context->Path());
 }
 
 /**
  @Status Stub
  @Notes
 */
 bool CGContextPathContainsPoint(CGContextRef context, CGPoint point, CGPathDrawingMode mode) {
-    NOISY_RETURN_IF_NULL(context, StubReturn());
-    UNIMPLEMENTED();
-    return StubReturn();
+    NOISY_RETURN_IF_NULL(context, false);
+
+    if (!context->HasPath()) {
```

Comment:
You really hate early returns, don't you?

---
In reply to: [85367729](https://github.com/Microsoft/WinObjC/pull/1240#discussion_r85367729) [](ancestors = 85367729)